text/0049-chain-ids.md

```diff
@@ -0,0 +1,234 @@
+  because attestation roots live in files.
+
+Another alternative: The Ethereum [JSON-RPC API](https://ethereum.org/en/developers/docs/apis/json-rpc/) returns the network id as a string, but in practice all the ids are actually integers.
+The mapping from network names to ids is available at [chainlist.org](https://chainlist.org/).
+
+* Assigning consecutive numbers instead of names creates additional work around an additional service and is somewhat unfriendly to ad-hoc dev networks and local networks.
+* One reviewer suggested that, instead of strings these should be 8-byte numbers, whose bit representation in ASCII is a human-readable string, so that no lookup service is needed.
+  However, this will actually double the size of these strings on the wire (and in block headers) in mainnet, since the string "main" is only 4 bytes.
+  The author feels that representing human-readable text using numbers is un-KISS.
+* Size on the wire is likely not a serious concern here. Each block contains at least three TxOut's, and including each fog hint and the key images, we will always be at about 1KB per block
+  at a minimum. Four to eight bytes is less than one percent of this, and this overhead should only be of concern in mainnet.
+
+Ethereum actually has two ids like this: A network id, and a chain id.
+
+* Peer-to-peer communication between nodes uses the network id
+* Transaction signatures use the chain id
+
+For most networks the two values are the same.
+Chain ID was introduced in [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) when the Ethereum Classic fork occurred, to prevent mixing of these two networks.
+
+* Putting the chain id within the transaction signature makes it easier for an off-line signer to
+  have visibility on what network the transaction is submitted to. If your transaction is submitted
+  via a proxy, the proxy would control which network id is expected.
+* In the status quo, the off-line signer has no visibility into this unless the proxy pipes an attested
+  connection through to them, which isn't possible for some types of off-line signers.
+
+In the Cosmos ecosystem, a [chain registry](https://github.com/cosmos/chain-registry) is hosted on github.
+Chains have both "names" which are human-readable strings, and "ids" which are also strings.
+
+Cosmos registry enforces some chain id [best practices](https://github.com/cosmos/cosmos-sdk/issues/5363).
+They expect chain-ids to match a pattern `[-a-zA-Z0-9]{3,47}`, otherwise they replace those chain-ids with a hash.
+This resembles the rules we proposed except that we are case-sensitive and only lower-case letters are allowed.
+
+# Prior art
+[prior-art]: #prior-art
+
+* [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md)
+* [Cosmos chain registry](https://github.com/cosmos/chain-registry)
+* [Cosmos chain id best practices](https://github.com/cosmos/cosmos-sdk/issues/5363)
+
```

Comment:
You could also reference that clients can incorporate the chain ID into their BIP44 key derivation schemes - as the example shows for Bitcoin TestNet

https://en.bitcoin.it/wiki/BIP_0044#Examples

Reply:
But I think in that case the chain id needs to be an int